Cesium.Parser/CParser.cs

```diff
@@ -90,7 +90,13 @@ public partial class CParser
         Expression function,
         IToken _,
         ArgumentExpressionList? arguments,
-        IToken __) => new FunctionCallExpression(function, arguments);
+        IToken __)
+    {
+        var castLike = function is ParenExpression { Contents: ConstantLiteralExpression } &&
```

Comment:
I think this should not be part of the parser, you can have that check in the lower phase too. so condition to have cast is to 
- have identifier in the paren (castLike)
- have identifier to be resolved to type
so if you have 2 checks in one place, this parameter would be not needed.

Reply:
The whole reason of having 'castLike' is to avoid ParenExpressions to appear in other parts of AST (they have no value by itself) - at this step we can check whether we have exactly one paren pair here and throw them away

Note that `((some_type)) (expr)` is no longer a valid cast